### PR TITLE
Breaking: ESP32 Core 3.0.0 only support

### PR DESCRIPTION
**Pull request by Jason2866**
## Description:

remove Platformio build support for Core 2.0.x and build all ESP32 variants only with Core 3.0.0 

CI is expected to fail. Changes in Github Actions.

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [x] The code change is tested and works with Tasmota core ESP32 V.3.0.0
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: .github/workflows/Tasmota_build_devel.yml [hunk 1]
```diff
@@ -96,14 +96,13 @@ jobs:
...
           - tasmota32solo1-safeboot
           - tasmota32-safeboot
-          - tasmota32c3-safeboot
+          - tasmota32c2-safeboot
           - tasmota32c3cdc-safeboot
           - tasmota32s2-safeboot
           - tasmota32s2cdc-safeboot
           - tasmota32s3-safeboot
           - tasmota32s3cdc-safeboot
           - tasmota32c2-safeboot
```
Comment (arendst) [inline .github/workflows/Tasmota_build_devel.yml:105]: Duplicate tasmota32c2-safeboot ???

File: .github/workflows/Tasmota_build_master.yml [hunk 2]
```diff
@@ -23,14 +23,13 @@ jobs:
...
           - tasmota32solo1-safeboot
           - tasmota32-safeboot
-          - tasmota32c3-safeboot
+          - tasmota32c2-safeboot
           - tasmota32c3cdc-safeboot
           - tasmota32s2-safeboot
           - tasmota32s2cdc-safeboot
           - tasmota32s3-safeboot
           - tasmota32s3cdc-safeboot
           - tasmota32c2-safeboot
```
Comment (arendst) [inline .github/workflows/Tasmota_build_master.yml:32]: Ditto

**Comment by arendst** [comment]
What is your opinion on changing ``..cdc.bin`` names to "normal" names without the cdc addition? As we now can detect HWCDC over serial, one binary is enough and I see no reason to call it ..cdc.bin

**Comment by Jason2866** [comment]
@arendst I had the same idea in mind. It is technically not 100% correct since the build settings from CDC is different to the non CDC builds. So I choose CDC for the name. And use the CDC build as one and only variant

**Comment by eku** [comment]
Since the update from 13.4 (tasmota32-display) to 14.0, my board no longer starts. With git-bisect I found out that this commit is the cause. It is a Heltic Wifi-Kit-32 V1 with ESP32-D0WDQ6. 

Should I open a ticket?

**Comment by arendst** [comment]
Did you read the changelog for v14.0? Did you notice the major change in display usage?

**Comment by eku** [comment]
> Did you read the changelog for v14.0? Did you notice the major change in display usage?

Sure, I did. #21183 does not break it.

With "no longer starts" I mean it does not start anymore, no output at serial, no WiFi connect. Nothing.

**Comment by arendst** [comment]
With no serial output it's time to call coroner or investigate if the correct binary is loaded....

**Comment by eku** [comment]
> investigate if the correct binary is loaded

At least I flashed the correct factory.bin file, the one from the release page.

And this is where I run out of options for further analysis.

**Comment by arendst** [comment]
Try to flash an older binary to see if you get serial back. If that works just perform an OTA of a newer release instead of serial factory.

**Comment by eku** [comment]
That worked, but it updates the app0 partition only. So the cause is in one of the other partitions, right?

**Comment by arendst** [comment]
If you did a serial flash of a factory image then all is rewritten not just app0. So I cannot understand your question. Using the factory image there is only one app0 partition and one safeboot partition.

**Comment by arendst** [comment]
Yes. Correct. If you want to update the safeboot partition execute command ``upload 2`` in the console. Using command ``restart 3`` switches between the two partitions.

**Pull request merged**